### ISSUE TITLE
Proprietary license

### ISSUE THREAD

**Issue by skr4n** [futo-org/voice-input#12]
The license used by the program I.e [FUTO Temporary License](https://gitlab.futo.org/alex/voiceinput/-/blob/master/FTL_LICENSE.md?ref_type=heads#section-2-grant-of-rights) does not adhere to the [Open Source definition](https://opensource.org/osd/) by restricting code modification and commercial redistribution.
I see that this is a temporary license and is intended to be replaced by a "more permissive" one in the future.
Im curious as to the requirements FUTO wanted as to make a custom license, Will the new license adhere to the Open source definition?
Why not use something like the GPL where the code is protected from being locked down, while staying open source, like you did with [Live Captions](https://github.com/abb128/LiveCaptions)?

**Comment by Quackdoc** [comment]
forwarded from the FUTO zulip, this was a reply I had gotten

```
Pavel Peganov / D-side: Quoting a response from Louis from "Letters to Louis" channel on Matrix:

    we planned on putting together a proper license at some point in the future
    that license was essentially scribbled on a napkin
    ah well. can't make everyone happy!

So I guess the answer is: emphasis on the "temporary".
Makes sense to me, permissive licensing is in a lot of ways a one-way trip: once you've allowed others to freely make modifications, they can take your work and reuse it a different project with no regulatory means of stopping it.
It can be both good and bad.
Good — in that bad execution with more exposure can twist the public perception of the core idea, which can hinder adoption of a genuinely good idea even when it's executed much better later.
Bad — in that it seriously restricts accountability in merit, because if the mainline goes bad, a circle of past contributors can't start a competing solution from the same codebase upholding what they perceived to be the mission of the project; they'll have to start "from plausible scratch" instead, which may be too much for the project to begin at all.

Pavel Peganov / D-side: Same source, earlier comments:

    We want to retain right to profit from it and also wanted to cut down on the bs that happened to newpipe where people put ads and trackers in it and out it on play store and outranked them

    Anytime I tell someone to download new pipe from f droid they say it's filled with ads because they ignore me and download from the play store a bs version from some scammer

    We're working on making alicense. It's a work in progress. We started with more restrictive because it's easier to make it less restrictive later than the opposite direction

```

**Comment by Cris-Edmundson** [comment]
I agree, restricting use of the same branding seems like a better way of dealing with forks attempting to pull one over on users who don't know the difference. The current licensing approach requires that the project always be managed in good faith until the end of time, which is nice to hope for, but not something you want to be depending on

**Comment by skr4n** [comment]
Totally agree with all the above comments.
FUTO has also used the same license for Grayjay, a video streaming. which I have also raised the same issue [here.](https://github.com/futo-org/grayjay-android/issues/18)

**Comment by h3ndrik** [comment]
Just wanted to say: Writing your own license is also a bad idea:

https://ben.balter.com/2016/08/01/why-you-shouldnt-write-your-own-open-source-license/

You kind of need to choose one of the free software licenses. Exchanging the temporary license for a custom license is probably not addressing any of the issues with that.

**Comment by milkey-mouse** [comment]
It's of course the developers' choice how to license their own work, but I'd really appreciate a license that at least allows modification. There's some changes I'd love to help out with, but as it stands I can't even modify the code for the sole purpose of submitting a merge request.

As a separate point, IMO the easiest way to meet [these requirements](https://github.com/futo-org/voice-input/issues/12#issuecomment-1759163448) for a license is some combination of GPL/MPL + enforced trademarks. Even GPL/MPL + some riders disallowing adding ads or removing a requirement to pay would be better than a custom license. Hell, GPL the app and only give the sources to paying customers. It works for Red Hat.

**Comment by h3ndrik** [comment]
> some combination [...] + some riders disallowing [...]

It'd be better not to add or combine anything to a license as this leads to [license incompatibility](https://en.wikipedia.org/wiki/License_compatibility) and makes it impossible to combine code. If in any way possible just use trademark law and force the people who fork it to rename their app so it doesn't get confused with FUTOs app. Or pick a different license that already contains the desired paragraphs.

**Comment by qrpnxz** [comment]
The [GNU General Public License](https://www.gnu.org/licenses/quick-guide-gplv3.html) is the state-of-the-art [copyleft](https://www.gnu.org/licenses/copyleft.html) license. Under the GPL, any work people publish based on VoiceInput would have to be released under the same license. It would require they provide all the source that corresponds to that version of the program, even the helper files that let you generate, install, and run that version. (e.g. gradle files)

With the GPL, it's not necessary to add anything regarding trademark rights because they are simply not granted.

But if one insists, version 3 explicitly has no issue with a copyright holder "declining to grant rights under trademark law for use of some trade names, trademarks, or service marks" per [section 7](https://www.gnu.org/licenses/gpl-3.0#section7)(e), and it shouldn't be difficult to avoid compatibility problems, because all the major licenses either say nothing about trademarks, or they also decline to grant those rights "except as may be necessary to comply with the notice requirements" or something to that effect.

The Mozilla Public License already declines to grant those rights in the text, but that license won't disallow people from releasing alternate versions of the program with proprietary extensions; that is, with additional work that you are not allowed to inspect or otherwise [use freely](https://www.gnu.org/philosophy/free-sw.html).

I hope VoiceInput is allowed to be used freely one day. In that case, I would recommend the GNU GPL because it guarantees those freedoms even when the program is modified.

**Comment by Lanchon** [comment]
my $0.02...

it is a matter of time: eventually someone will make a free version of this app and this app will disappear into oblivion.

precisely because this app targets freedom and privacy valuing users, most will switch to a freer alternative when available, even if it originally is less polished. polish will come with users contributing at some later time.

this is exactly what happened to minix: minix was posed to take over the world, it was at the right time and place, but unfortunately it had a non-free license that excluded commercial use. so linux was born. initially it was extremely less usable and useful than minix, just a PoC. but it was free, so it grew... and now it is an extremely polished piece of top-noch engineering that runs the world. later minix was freed, but it was too late.

minix could have ruled the world. but now it is just an obscure piece of history with no relevancy. (except within intel, which coincidentally is a company heading to irrelevancy itself.)

**Comment by Danny3** [comment]
Voice input can also be done by other apps that are fully open source and available in the main F-droid repository, as I already explained here:
https://github.com/futo-org/grayjay-android/issues/18#issuecomment-3944393199
